src/LinkedList.ts

```diff
@@ -102,4 +98,9 @@ export class LinkedList {
             throw Error("Only member pointers names 'next' are supported right now.");
         }
     }
+
+    public getNodeIdAt(pointer: string): number {
+        return this.getPointerFromString(pointer).deref()?.getId();
```

Comment:
Its called optional chaining and its to avoid a `null` error when calling `.getID()`. 

Read more here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining. 

Its pretty new in JS, but the Typescript compiler can compile it out if we want to target older JS.

Reply:
For your entertainment, this code is compiled to the following JavaScript: 

```js
getNodeIdAt(pointer) {
        var _a;
        return (_a = this.getPointerFromString(pointer).deref()) === null || _a === void 0 ? void 0 : _a.getId();
    }
```